examples/cpp/dds/BasicConfigurationExample/BasicConfigurationPublisher.cpp

```diff
@@ -78,24 +79,53 @@ bool HelloWorldPublisher::init(
 
     // TRANSPORT CONFIG
     // If it is set, not use default and set the transport
-    if (transport != DEFAULT)
+    if (transport != DEFAULT || hops > 0 )
     {
         pqos.transport().use_builtin_transports = false;
 
-        if (transport == SHM)
-        {
-            auto shm_transport = std::make_shared<SharedMemTransportDescriptor>();
-            pqos.transport().user_transports.push_back(shm_transport);
-        }
-        else if (transport == UDPv4)
+        switch ( transport )
         {
-            auto udp_transport = std::make_shared<UDPv4TransportDescriptor>();
-            pqos.transport().user_transports.push_back(udp_transport);
+            case SHM:
+            {
+                auto shm_transport = std::make_shared<SharedMemTransportDescriptor>();
+                pqos.transport().user_transports.push_back(shm_transport);
+            }
+            break;
+            case UDPv4:
+            {
+                auto udp_transport = std::make_shared<UDPv4TransportDescriptor>();
+                pqos.transport().user_transports.push_back(udp_transport);
+            }
+            break;
+            case UDPv6:
+            {
+                auto udp_transport = std::make_shared<UDPv6TransportDescriptor>();
+                pqos.transport().user_transports.push_back(udp_transport);
+            }
+            break;
+            case DEFAULT:
+            default:
+            {
+                // mimick default transport selection
```

Comment:
Isn't it possible to directly modify the builtin transports?

Reply:
nope